x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_config_panel.tsx

```diff
@@ -72,258 +75,187 @@ export const chartTypeIcons: Array<{ id: SeriesType; label: string; iconType: Ic
-  const { state, datasource, setState } = props;
-
-  return (
-    <EuiForm className="lnsConfigPanel">
-      <EuiFormRow
-        label={i18n.translate('xpack.lens.xyChart.chartTypeLabel', {
-          defaultMessage: 'Chart type',
-        })}
-      >
-        <EuiButtonGroup
-          legend={i18n.translate('xpack.lens.xyChart.chartTypeLegend', {
-            defaultMessage: 'Chart type',
-          })}
-          name="chartType"
-          className="eui-displayInlineBlock"
-          data-test-subj="lnsXY_seriesType"
-          options={chartTypeIcons.map(type =>
-            type.id.includes('stacked') && state.splitSeriesAccessors.length === 0
-              ? { ...type, isDisabled: true }
-              : type
-          )}
-          idSelected={state.seriesType}
-          onChange={seriesType => {
-            const isHorizontal = seriesType === 'horizontal_bar';
-            setState({
-              ...state,
-              seriesType: seriesType as SeriesType,
-              x: {
-                ...state.x,
-                position: isHorizontal ? Position.Left : Position.Bottom,
-              },
-              y: {
-                ...state.y,
-                position: isHorizontal ? Position.Bottom : Position.Left,
-              },
-            });
-          }}
-          isIconOnly
-        />
-      </EuiFormRow>
+function updateLayer(state: State, layer: UnwrapArray<State['layers']>, index: number): State {
+  const newLayers = [...state.layers];
+  newLayers[index] = layer;
 
-      <EuiFormRow>
-        <EuiSwitch
-          label={i18n.translate('xpack.lens.xyChart.showLegendLabel', {
-            defaultMessage: 'Show legend',
-          })}
-          checked={state.legend.isVisible}
-          data-test-subj="lnsXY_legendIsVisible"
-          onChange={() =>
-            setState({
-              ...state,
-              legend: { ...state.legend, isVisible: !state.legend.isVisible },
-            })
-          }
-        />
-      </EuiFormRow>
+  return {
+    ...state,
+    layers: newLayers,
+  };
+}
 
-      {state.legend.isVisible && (
-        <EuiFormRow
-          label={i18n.translate('xpack.lens.xyChart.legendPositionLabel', {
-            defaultMessage: 'Legend position',
-          })}
-        >
-          <EuiButtonGroup
-            legend={i18n.translate('xpack.lens.xyChart.legendPositionButtonLabel', {
-              defaultMessage: 'Legend position',
-            })}
-            data-test-subj="lnsXY_legendPosition"
-            name="legendPosition"
-            options={positionIcons}
-            idSelected={state.legend.position}
-            onChange={position =>
-              setState({ ...state, legend: { ...state.legend, position: position as Position } })
-            }
-            isIconOnly
-          />
-        </EuiFormRow>
-      )}
+function newLayerState(layerId: string): LayerConfig {
+  return {
+    layerId,
+    datasourceId: 'indexpattern', // TODO: Don't hard code
```

Comment:
not necessary anymore

Reply:
Not sure what you're suggesting instead?

Reply:
It's apparently gone:

```js

export type LayerConfig = AxisConfig & {
  layerId: string;
  xAccessor: string;
  accessors: string[];
  seriesType: SeriesType;
  splitAccessor: string;
};
```

I guess it's being tracked internally in the editor frame which kinda makes sense.

Reply:
That was my point Chris - resolved 👍